google/cloud/storage/client_test.cc

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/client.h"
+#include "google/cloud/storage/retry_policy.h"
+#include "google/cloud/storage/testing/canonical_errors.h"
+#include "google/cloud/storage/testing/mock_client.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+using namespace testing::canonical_errors;
+namespace {
+using namespace ::testing;
+
+class ObservableRetryPolicy : public LimitedErrorCountRetryPolicy {
+ public:
+  using LimitedErrorCountRetryPolicy::LimitedErrorCountRetryPolicy;
+
+  std::unique_ptr<RetryPolicy> clone() const override {
+    return std::unique_ptr<RetryPolicy>(new ObservableRetryPolicy(*this));
+  }
+
+  bool IsExhausted() const override {
+    ++is_exhausted_count;
```

Comment:
Naming nit: Can you call this `is_exhausted_call_count`?  At first glance, `is_exhausted_count` reads as if this were the number of times `IsExhausted` returned true.

Reply:
Done.